File: backend/src/main/java/dev/laszlo/dto/StoryDto.java
package dev.laszlo.dto;

import java.time.LocalDateTime;

/**
 * Response DTO for story data
 */
public class StoryDto {

    private Long id;
    private String storyId;
    private String title;
    private String description;
    private String coverImageUrl;
    private boolean isPublished;
    private Long createdByUserId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    // Constructors
    public StoryDto() {}

    public StoryDto(Long id, String storyId, String title, String description,
                    String coverImageUrl, boolean isPublished, Long createdByUserId,
                    LocalDateTime createdAt, LocalDateTime updatedAt) {
        this.id = id;
        this.storyId = storyId;
        this.title = title;
        this.description = description;
        this.coverImageUrl = coverImageUrl;
        this.isPublished = isPublished;
        this.createdByUserId = createdByUserId;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getStoryId() { return storyId; }
    public void setStoryId(String storyId) { this.storyId = storyId; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getCoverImageUrl() { return coverImageUrl; }
    public void setCoverImageUrl(String coverImageUrl) { this.coverImageUrl = coverImageUrl; }

    public boolean isPublished() { return isPublished; }
    public void setPublished(boolean published) { isPublished = published; }

    public Long getCreatedByUserId() { return createdByUserId; }
    public void setCreatedByUserId(Long createdByUserId) { this.createdByUserId = createdByUserId; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
}

File: frontend/lib/models/admin/story_dto.dart
class StoryDto {
  final int id;
  final String storyId;
  final String title;
  final String? description;
  final String? coverImageUrl;
  final int createdByUserId;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool published;

  StoryDto({
    required this.id,
    required this.storyId,
    required this.title,
    this.description,
    this.coverImageUrl,
    required this.createdByUserId,
    required this.createdAt,
    required this.updatedAt,
    required this.published,
  });

  factory StoryDto.fromJson(Map<String, dynamic> json) {
    return StoryDto(
      id: json['id'] as int,
      storyId: json['storyId'] as String,
      title: json['title'] as String,
      description: json['description'] as String?,
      coverImageUrl: json['coverImageUrl'] as String?,
      createdByUserId: json['createdByUserId'] as int,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
      published: json['published'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'storyId': storyId,
      'title': title,
      'description': description,
      'coverImageUrl': coverImageUrl,
      'createdByUserId': createdByUserId,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
      'published': published,
    };
  }

  StoryDto copyWith({
    int? id,
    String? storyId,
    String? title,
    String? description,
    String? coverImageUrl,
    int? createdByUserId,
    DateTime? createdAt,
    DateTime? updatedAt,
    bool? published,
  }) {
    return StoryDto(
      id: id ?? this.id,
      storyId: storyId ?? this.storyId,
      title: title ?? this.title,
      description: description ?? this.description,
      coverImageUrl: coverImageUrl ?? this.coverImageUrl,
      createdByUserId: createdByUserId ?? this.createdByUserId,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      published: published ?? this.published,
    );
  }
}
